base/interrupts.jl

```diff
@@ -0,0 +1,232 @@
+# Internal methods, only to be used to change to a different global interrupt handler
+function _register_global_interrupt_handler(handler::Task)
+    handler_ptr = Base.pointer_from_objref(handler)
+    slot_ptr = cglobal(:jl_interrupt_handler, Ptr{Cvoid})
+    Intrinsics.atomic_pointerset(slot_ptr, handler_ptr, :release)
+end
+function _unregister_global_interrupt_handler()
+    slot_ptr = cglobal(:jl_interrupt_handler, Ptr{Cvoid})
+    Intrinsics.atomic_pointerset(slot_ptr, C_NULL, :release)
+end
+
+const INTERRUPT_HANDLERS_LOCK = Threads.ReentrantLock()
+const INTERRUPT_HANDLERS = Dict{Module,Vector{Task}}()
+const INTERRUPT_HANDLER_RUNNING = Threads.Atomic{Bool}(false)
```

Comment:
Aren't these atomics deprecated?

Reply:
What would you use instead? The `@atomics` interface doesn't provide an alternative for a single but entirely atomically accessed resource, as far as I'm aware.

Reply:
I can turn it into a struct, but there's only ever one instance of it.

Reply:
We should make it just a global, but currently those don't support custom atomic annotations (they are always just release/consume)

Reply:
The "clean" solution would be to have a struct with `@atomic` fields, stored in a global `Ref`. The constructor would check if the `Ref` is already assigned and refuse being instantiated otherwise.